e2etests/test/empty_messages.js

```diff
@@ -0,0 +1,210 @@
+    });
+
+    it('Service sends a C2D message with an empty payload and it is received by the device', function (done) {
+      var deviceClientParticipant = 'deviceClient';
+      var serviceClientParticipant = 'serviceClient';
+      var testRendezvous = new Rendezvous(done);
+      testRendezvous.imIn(deviceClientParticipant);
+      this.timeout(18000);
+      //
+      // The size of non payload is calculated based on the value of the messageId value size,
+      // the expiryTimeUtc value size and then on the application property value size plus one
+      // for the application property name.
+      //
+      var uuidData = uuid.v4();
+      var message = new Message('');
+      message.messageId = uuidData;
+      message.expiryTimeUtc = Date.now() + 60000; // Expire 60s from now, to reduce the chance of us hitting the 50-message limit on the IoT Hub
+
+      var foundTheMessage = false;
+      deviceClient.open(function (openErr) {
+        debug('device has opened.');
+        if (openErr) {
+          done(openErr);
+        } else {
+          debug('about to connect a listener.');
+          deviceClient.on('message', function (msg) {
+            debug('received a message');
+            //
+            // Make sure that the message we are looking at is one of the messages that we just sent.
+            //
+            if (msg.messageId === uuidData) {
+              foundTheMessage = true;
+            }
+            //
+            // It doesn't matter whether this was a message we want, complete it so that the message queue stays clean.
+            //
+            deviceClient.complete(msg, function (err, result) {
+              if (err) {
+                done(err);
+              } else {
+                assert.equal(result.constructor.name, 'MessageCompleted');
+                if (foundTheMessage) {
+                  deviceClient.removeAllListeners('message');
+                  testRendezvous.imDone(deviceClientParticipant);
+                }
+              }
+            });
+          });
+          debug('about to open the service client');
+          serviceClient.open(function (serviceErr) {
+            debug('At service client open callback - error is:' + serviceErr);
+            if (serviceErr) {
+              done(serviceErr);
+            } else {
+              testRendezvous.imIn(serviceClientParticipant);
+              serviceClient.send(provisionedDevice.deviceId, message, function (sendErr) {
+                debug('At service client send callback - error is: ' + sendErr);
+                if (sendErr) {
+                  done(sendErr);
+                } else {
+                  testRendezvous.imDone(serviceClientParticipant);
+                }
+              });
+            }
+          });
+        }
+      });
+    });
+  });
+
+  describe('Over ' + deviceTransport.name + ' using device/eventhub clients - messaging', function () {
+    this.timeout(18000);
```

Comment:
>18000 [](start = 17, length = 5)

same here

Reply:
Fixed.

---
In reply to: [227565298](https://github.com/Azure/azure-iot-sdk-node/pull/401#discussion_r227565298) [](ancestors = 227565298)